central/scannerdefinitions/handler/handler.go

```diff
@@ -461,6 +483,60 @@ func (h *httpHandler) openMostRecentV4File(t updaterType, updaterKey, fileName s
 	return file, nil
 }
 
+// openMostRecentV4OfflineFile gets desired offline file from compressed bundle: offlineScannerV4DefinitionBlobName
+func (h *httpHandler) openMostRecentV4OfflineFile(ctx context.Context, t updaterType, updaterKey, fileName string) (file *vulDefFile, err error) {
+	log.Debugf("Getting v4 offline data for updater key: %s", updaterKey)
+	openedFile, err := h.openOfflineBlob(ctx, offlineScannerV4DefinitionBlobName)
+	if err == nil && openedFile == nil {
+		log.Warnf("Blob %s does not exist", offlineScannerV4DefinitionBlobName)
+		return nil, errors.New("No valid scanner V4 file in offline mode")
+	}
+
+	var offlineFile *vulDefFile
+	defer utils.IgnoreError(openedFile.Close)
+	switch t {
+	case mappingUpdaterType:
+		// search mapping file
+		fileName = path.Base(fileName)
+		targetFile, err := openFromArchive(openedFile.Name(), fileName)
+		if err != nil {
+			return nil, err
+		}
+		offlineFile = &vulDefFile{File: targetFile, modTime: openedFile.modTime}
+	case vulnerabilityUpdaterType:
+		// check version information in manifest
+		mf, err := openFromArchive(openedFile.Name(), "manifest.json")
+		if err != nil {
+			return nil, err
+		}
+
+		offlineV, err := getOfflineFileVersion(mf)
+		if err != nil {
+			return nil, err
+		}
+		defer utils.IgnoreError(mf.Close)
+
+		if offlineV != majorVersion.FindString(updaterKey) {
+			msg := fmt.Sprintf("failed to get offline vuln file, uploaded file is version: %s and requested file version is: %s", offlineV, updaterKey)
+			log.Errorf(msg)
+			return nil, errors.New(msg)
+		}
+
+		vulns, err := openFromArchive(openedFile.Name(), "vulns.json.zst")
+		if err != nil {
+			return nil, err
+		}
+		offlineFile = &vulDefFile{File: vulns, modTime: openedFile.modTime}
+	default:
+		return nil, fmt.Errorf("unknown Scanner V4 updater type: %s", t)
+	}
+
+	// not closing offlineFile here
+	// as file will be closed in getV4() after http serves content
```

Comment:
no need for this comment. The function returns the file, so it'd be expected to be opened